paladin/mod.ts

```diff
@@ -0,0 +1,203 @@
+    "X-Frame-Options": "SAMEORIGIN",
+    "X-DNS-Prefetch-Control": false,
+  };
+
+  /**
+   * The middleware function that's called by Drash.
+   *
+   * @param request - The request object.
+   * @param response - (optional) The response object.
+   */
+  function paladin(
+    request: Drash.Http.Request,
+    response?: Drash.Http.Response,
+  ): void {
+    // If there is a response, then we know this is occurring after the request
+    if (response) {
+      // Set "X-XSS-Protection" header. See https://helmetjs.github.io/docs/xss-filter/
+      if (configs["X-XSS-Protection"] !== false) {
+        response.headers.set(
+          "X-XSS-Protection",
+          defaultConfigs["X-XSS-Protection"],
+        );
+        configs["X-XSS-Protection"] = true;
+      }
+
+      // Set "Referrer-Policy" header if passed in. See https://helmetjs.github.io/docs/referrer-policy/
+      if (configs["Referrer-Policy"]) {
+        response.headers.set(
+          "Referrer-Policy",
+          configs["Referrer-Policy"].toString(),
+        );
+      }
+
+      // Set the "X-Content-Type-Options" header. See https://helmetjs.github.io/docs/dont-sniff-mimetype/
+      if (configs["X-Content-Type-Options"] !== false) {
+        response.headers.set(
+          "X-Content-Type-Options",
+          defaultConfigs["X-Content-Type-Options"],
+        );
+        configs["X-Content-Type-Options"] = true;
+      }
+
+      // Set the "Strict-Transport-Security" header. See https://helmetjs.github.io/docs/hsts/
+      let hstsHeader = "";
+      if (configs.hsts!.max_age) { // if set to a number
+        hstsHeader += "max-age=" + configs.hsts!.max_age;
+      } else if (configs.hsts!.max_age !== false) { // not disabled
+        hstsHeader += "max-age=" + defaultConfigs.hsts.max_age;
+      }
+      if (hstsHeader && configs.hsts!.include_sub_domains === true) {
+        hstsHeader += "; include_sub_domains";
+      } else if (hstsHeader && configs.hsts!.include_sub_domains !== false) {
+        hstsHeader += "; include_sub_domains";
+      }
+      if (hstsHeader && configs.hsts!.preload === true) {
+        hstsHeader += "; preload";
+      }
+      // } else if (hstsHeader && configs.hsts!.preload !== false) {
+      //   hstsHeader += "; preload"
+      // }
+      if (hstsHeader) {
+        response.headers.set("Strict-Transport-Security", hstsHeader);
+      }
+      configs.hsts = {
+        preload: configs.hsts!.preload ?? false,
+        max_age: configs.hsts!.max_age ?? defaultConfigs.hsts.max_age,
+        include_sub_domains: !!configs.hsts!.include_sub_domains ??
+          !!defaultConfigs.hsts.include_sub_domains,
+      };
+
+      // Delete or modify the "X-Powered-By" header. See https://helmetjs.github.io/docs/hide-powered-by/
+      if (typeof configs["X-Powered-By"] === "string") { // user wants to modify the header
+        response.headers.set(
+          "X-Powered-By",
+          configs["X-Powered-By"].toString(),
```

Comment:
do we need to call toString() if we already checked if its a string?

Reply:
yeah that was annoying, the tsc was complaining because a type `string | boolean` could not be assigned to `string`, but as you can see in the if statement, we know it's a string - any ideas of how to combat that?

Reply:
no ideas to combat that